nix/nix.go

```diff
@@ -360,7 +361,13 @@ func Push(ctx *ssh.SSHContext, host Host, paths ...string) (err error) {
 		keyArg = "?ssh-key=" + ctx.IdentityFile
 	}
 	if ctx.SkipHostKeyCheck {
-		env = append(env, fmt.Sprintf("NIX_SSHOPTS=%s", "-o StrictHostkeyChecking=No -o UserKnownHostsFile=/dev/null"))
+  		sshOpts = append(sshOpts, fmt.Sprintf("%s", "-o StrictHostkeyChecking=No -o UserKnownHostsFile=/dev/null"))
+	}
+	if ctx.ConfigFile != "" {
+  		sshOpts = append(sshOpts, fmt.Sprintf("-F %s", ctx.ConfigFile))
```

Comment:
Can thing config file be an output of a nix builder or does it have to exist before the deployment is started?

Reply:
I don't see why not. As long as it returns a path and `ssh` can access that path when passing `-F /path`, should be fine.